### ISSUE TITLE
Missing keyword handling for IBM DB2 dialect

### ISSUE THREAD

**Issue by szapp** [sqlfluff/sqlfluff#8248]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The db2 dialect seems to be missing handling of some keywords. I experience parsing erros that interrupt linting of the remaining SQL script.

Suppression with `-- noqa: PRS` does not help, because formatting is interrupted.

I encounter parsing errors on these valid IBM DB2 statements:

```sql
comment on column schema.table.column is 'hello world';
```

```sql
-- create view schema.foo
-- ...
with no row movement;
```

Several user grants in one statement:

```sql
grant all
on schema.table
to  user user1,
    user user2,
    user user3;
```

From a brief look into the source code, I suspect the class `CommentOnStatementSegment` might be required for the comment-clause as is present in the dialects of postgres and databricks.

### Expected Behaviour

The above statements are parsed and recognized for the db2 dialect.

### Observed Behaviour

1. This file content results in the following linting error:

```sql
create view schema.foo as
select col1
from schema.bar
with no row movement;
```

```sh
$> sqlfluff lint testing_with_no_row_movement.sql --dialect db2
== [testing_with_no_row_movement.sql] FAIL
L:   4 | P:   1 | AL01 | Implicit/explicit aliasing of table.
                       | [aliasing.table]
L:   4 | P:   1 | AL05 | Alias 'with' is never used in SELECT statement.
                       | [aliasing.unused]
L:   4 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:   4 | P:   1 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
L:   4 | P:   5 | LT02 | Expected line break and no indent before 'no'.
                       | [layout.indent]
L:   4 | P:   6 |  PRS | Line 4, Position 6: Found unparsable section: 'no row
                       | movement'
WARNING: Parsing errors found and dialect is set to 'db2'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

2. This file content results in the following linting error:

```sql
comment on column schema.table.column is 'hello';
```

```sh
$> sqlfluff lint testing_comment_on_column.sql --dialect db2
== [testing_comment_on_column.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'comment
                       | on column schema.table.column is...'
WARNING: Parsing errors found and dialect is set to 'db2'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

3. This file content results in the following linting error:

```sql
grant all
on schema.table
to  user user1,
    user user2,
    user user3;
```

```sh
$> sqlfluff lint testing_grant_multi_user.sql --dialect db2
== [testing_grant_multi_user.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'grant
                       | all\non schema.table\nto  user user1...'
WARNING: Parsing errors found and dialect is set to 'db2'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

See **Observed Behaviour**.

### Dialect

db2

### Version

sqlfluff v4.2.2 [CPython 3.14.6]

### Configuration

None. Minimal example only specifiying the dialect as db2.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
